### DISCUSSION TITLE
Removing dups in repeating field

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#4956] [Q&A]
Looking for a good way to compare provider names and remove duplicates. In the below example in OBR-16, the .2 and .3 are the same. Not concerned about the .1 ID.

`24141^GOLLAPUDI^SUNIL^K.^^^^D4D_PP~682070^GOLLAPUDI^SUNIL^K.^^^^D4D_PP`

Thought of adding them all to an array and comparing each entry, wondering if there is a smarter way to do this.

**Comment by MichaelLeeHobbs** [comment]
Add them to a Map or Set where the key is Last+First+Middle. In your sample data they have different ID's so maybe use a map and store the multiple IDs.

**Reply by pacmano1** [reply]
I might use XSLT for this since you are working in e4x anyway see https://stackoverflow.com/questions/10912544/removing-duplicate-elements-with-xslt

**Comment by jonbartels** [comment]
https://github.com/nextgenhealthcare/connect-examples may have an example. 

https://github.com/nextgenhealthcare/connect-examples/tree/master/Code%20Templates/Filter%20XMLLists may be relevant.

**Comment by tonygermano** [comment]
How do you determine which one you need to keep after finding a duplicate? Like @MichaelLeeHobbs said, the IDs are different.
